docs/Course Wiki/CS Course Offerings/CS374A.md

```diff
@@ -0,0 +1,95 @@
+
+### Exams and Grading
+
+CS374A has two midterms and one final, and typically does not curve exam scores. Exam scores make up roughly 70% of a student's grade, and the remainder comes from homework scores. The lowest 4-6 homework problem scores over the semester are not included in grade calculation, and one GPS is considered equal to one homework problem.
+
+## Instructors
+
+The instructors for CS374A rotate between CS theory staff:
+
+- Sariel Har-Peled: considered to be the most difficult CS374A professor, but usually writes interesting problems and can be a fun lecturer
+- Jeff Erickson: author of the CS374 textbook and an engaging lecturer, some students wait for him to be teaching before taking the course
+- Chandra Chekuri: an engaging lecturer that cares about students and writes reasonable exams
+- Ruta Mehta and Timothy Chan: usually co-teaching with Professor Mehta lecturing, strong lectures but homework and exam questions may be difficult
+
+## Course Tips
+
+It is important to work on homework assignments as a group, with others you work well with. Some groups prefer to solve problems together, while others prefer to break up the problems between members. Whichever approach you take, make sure to understand the solutions to every homework problem, because they may contain techniques that will be useful on exams.
+
+You can save several hours on homework assignments by attending discussion sections for two hours per week. Since discussion attendance is optional, you can go to a different discussion section you are registered if there's space. Discussion sections are also a good place to meet potential groupmates for homework assignments.
+
+It is important to show your work on all assignments, and if you handwrite, write legibly. Graders are generous with partial credit.
+
+## Life After
+
+CS374 is a prerequisite for [CS421 (Programming Languages)](./CS421.md), a required course for Computer Science majors. Students who enjoyed the middle third of the course will enjoy [CS473 (Algorithms)](./CS473.md). Students who enjoyed the first and last thirds of the course will enjoy [CS475 (Formal Models of Computation)](./CS475.md).
+
+## Infamous Topics
+
+- Language transformations: there is usually a language transformations question on the first midterm, and this is an unintuitive topic. TAs and CAs are good resources to ask for help, and Professor Erickson's textbook contains several examples.
+- Dynamic programming: dynamic programming solutions rely on recursion, which can be unfamiliar. The best way to improve is practice.
+- NP-completeness reductions: it is easy to prove the wrong direction, so you should draw diagrams and examples before starting your proof.
+
+## Resources
+
+- [Professor Erickson's Textbook](https://algorithms.wtf)
+- [Professor Erickson's Lectures](https://www.youtube.com/watch?v=A289028I8tc&list=PL0v718LJg-7_4Zwx3CE7kZ398mhlB2TqF)
+- [LaTeX DFA generator](https://madebyevan.com/fsm/)
```

Comment:
Write that this is useful for the first few homeworks so people know why this is here